### PR TITLE
Update index.d.ts

### PR DESCRIPTION
**Pull request by BlackGlory**
Update `index.d.ts` to v3.0

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by StarpTech — APPROVED** [review]
lgtm, thanks.

**Review by mcollina — CHANGES_REQUESTED** [review]
Would you mind to add a test for this type?

**Comment by BlackGlory** [comment]
@mcollina I did not find a file for testing TypeScript types in the repo, `test/index.test-d.ts` doesn't look like a unit test.

**Comment by BlackGlory** [comment]
`test/index.test-d.ts` is the v2.0 test file, so I updated it to v3.0.
But IMHO, since the file is not updated with the new version, obviously these tests are not helpful.

**Comment by BlackGlory** [comment]
Considering the properties of the field type, it seems necessary to split Multipart into two interfaces.
But it will break the examples in the README, because we need to distinguish between different types. I'm not sure if these definitions should be added.

```ts
type Multipart = MultipartField | MultipartFile

interface MultipartField {
  fieldname: string
  value: string
  fieldnameTruncated: boolean
  valueTruncated: boolean
  fields: MultipartFields
}

interface MultipartFile  {
  toBuffer: () => Promise<Buffer>,
  file: NodeJS.ReadableStream,
  filepath: string,
  fieldname: string,
  filename: string,
  encoding: string,
  mimetype: string,
  fields: MultipartFields
}
```

**Comment by hixus** [comment]
I might also export all interfaces (for example Multipart) so one can use them for own function declarations that use the result

**Comment by mcollina** [comment]
I don't understand the question. Is this ready to land?

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by hixus** [comment]
> I don't understand the question. Is this ready to land?

Could be different pr, but for example I might wan't to define separate handler which would take just the Multipart as an argument. Currently not possible without defining it again.

```
const handleFile = async (part: Multipart) => {
  pump(part.file, fs.createWriteStream(part.filename))
}

server.post("/upload", async (req, reply) => {
  const parts = req.parts()
  for await (const part of parts) {
    if (part.file) {
      await handleFile(part)
    } else {
      console.log(part)
    }
  }
  reply.send()
});
```

**Comment by BlackGlory** [comment]
This PR is ready to merge.

It may take someone more familiar with this project to decide which TypeScript interfaces should be exported, not me.

File: index.d.ts [hunk 1]
```diff
@@ -18,8 +19,8 @@ interface BodyEntry {
     limit: false
 }
 
-interface MultipartFields {
-  [x: string]: Multipart;
+export interface MultipartFields {
+    [x: string]: Multipart | Multipart[];
 }
 
 interface Multipart {
```
Comment (fox1t) [inline index.d.ts:26]:
```suggestion
export interface Multipart {
```

**Review by fox1t — CHANGES_REQUESTED** [review]

**Comment by fox1t** [comment]
Nice work! Thanks for you contribution. 
There is one missing export to add, then this PR can be merged.

**Review by fox1t — APPROVED** [review]
LGTM

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**